lib/elixir/lib/uri.ex

```diff
@@ -699,3 +699,9 @@ defimpl String.Chars, for: URI do
     ]
   end
 end
+
+defimpl Inspect, for: URI do
+  def inspect(uri, _) do
+    "~URI<" <> URI.to_string(uri) <> ">"
```

Comment:
there's a problem with this sigil that I am unsure how to solve:

```elixir
iex(1)> URI.parse("https://elixir-lang.org?v=>")
~URI<https://elixir-lang.org?v=>>

iex(2)> [ ~URI<https://elixir-lang.org?v=>> ]
** (SyntaxError) iex:2: syntax error before: ']'
```

I guess we should escape `<` as `\<` in the output but maybe it'd be weird?